data/option.go

```diff
@@ -76,6 +77,57 @@ func (m microVec3) Type() string {
 	return "microVec3"
 }
 
+// FanSpeedOptions used to control fan speed at given layers.
+type FanSpeedOptions struct {
+  LayerToSpeedLUT map[int]int
+}
+
+// NewDefaultFanSpeedOptions Creates instance FanSpeedOptions
+// and sets a of full fan (255) at layer 3.
+func NewDefaultFanSpeedOptions() FanSpeedOptions {
+	fo := FanSpeedOptions{}
+	fo.LayerToSpeedLUT = make(map[int]int)
+	fo.LayerToSpeedLUT[2] = 255
+	return fo
+}
+
+func (f FanSpeedOptions) Type() string {
+	return "FanSpeedOptions"
+}
+
+func (f FanSpeedOptions) String() string {
+	var s []string
+	for k,v := range f.LayerToSpeedLUT {
+		s = append(s, fmt.Sprintf("%d=%d",k,v))
+	}
+	return strings.Join(s,",")
+}
+
+// Set takes string in format layerNo2=FanSpeed2,LayerNo2=FanSpeed2
+// Checks fan speed is within allowed range 0-255.
+// Also confirms layer is at at least 0 or above.
+func (f *FanSpeedOptions) Set(s string) error {
```

Comment:
I was looking for tests but didn't see any earlier. I notice with the latest master there is, will add some in soon. Yesterday when I pulled from upstream master I didn't even notice the warning where it said since I didn't specify a branch (and no default set) that it didn't bother pulling anything. doh! :)

Reply:
added several tests to check defaults, layer numbers, speed numbers.